File: src/features/organizations/dto/list-organizations.dto.ts
import { Transform } from 'class-transformer';
import { IsBoolean, IsIn, IsInt, IsOptional, IsString, Max, Min } from 'class-validator';

export class ListOrganizationsDto {
  @IsOptional()
  @IsString()
  search?: string;

  @IsOptional()
  @Transform(({ value }) => value === true || value === 'true' || value === '1')
  @IsBoolean()
  includeArchived?: boolean;

  @IsOptional()
  @IsIn(['shared', 'dedicated'])
  datastoreType?: 'shared' | 'dedicated';

  @IsOptional()
  @Transform(({ value }) => value === true || value === 'true' || value === '1')
  @IsBoolean()
  legalHold?: boolean;

  @IsOptional()
  @Transform(({ value }) => value === true || value === 'true' || value === '1')
  @IsBoolean()
  piiStripped?: boolean;

  @IsOptional()
  @Transform(({ value }) => (value === undefined ? undefined : Number(value)))
  @IsInt()
  @Min(1)
  page?: number;

  @IsOptional()
  @Transform(({ value }) => (value === undefined ? undefined : Number(value)))
  @IsInt()
  @Min(1)
  @Max(100)
  limit?: number;
}

File: spring-boot-api/src/main/java/com/mytegroup/api/dto/organizations/ListOrganizationsDto.java
package com.mytegroup.api.dto.organizations;

import com.mytegroup.api.entity.enums.organization.DatastoreType;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;

public record ListOrganizationsDto(
    String search,
    Boolean includeArchived,
    DatastoreType datastoreType,
    Boolean legalHold,
    Boolean piiStripped,
    @Min(value = 1, message = "Page must be at least 1")
    Integer page,
    @Min(value = 1, message = "Limit must be at least 1")
    @Max(value = 100, message = "Limit must be at most 100")
    Integer limit
) {
    public ListOrganizationsDto {
        if (search != null) {
            search = search.trim();
        }
    }
}



